receiver/awsxrayreceiver/internal/translator/sql.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator
+
+import (
+	"fmt"
+	"regexp"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsxrayreceiver/internal/tracesegment"
+)
+
+func addSQLToSpan(sql *tracesegment.SQLData, attrs *pdata.AttributeMap) error {
+	if sql == nil {
+		return nil
+	}
+
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/exporter/awsxrayexporter/translator/sql.go#L33
+	if sql.URL != nil {
+		dbURL, dbName, err := splitSQLURL(*sql.URL)
+		if err != nil {
+			return err
+		}
+		attrs.UpsertString(conventions.AttributeDBConnectionString, dbURL)
+		attrs.UpsertString(conventions.AttributeDBName, dbName)
+	}
+	// not handling sql.ConnectionString for now because the X-Ray exporter
+	// does not support it
+	addString(sql.DatabaseType, conventions.AttributeDBSystem, attrs)
+	addString(sql.SanitizedQuery, conventions.AttributeDBStatement, attrs)
+	addString(sql.User, conventions.AttributeDBUser, attrs)
+	return nil
+}
+
+var re = regexp.MustCompile(`^(.+\/\/.+)\/(.+)$`)
+
+const (
+	dbURLI  = 1
+	dbNameI = 2
+)
+
+func splitSQLURL(rawURL string) (string, string, error) {
+	m := re.FindStringSubmatch(rawURL)
```

Comment:
I think you can just use `strings.LastIndexByte(rawURL, '/')` instead of the regex

Reply:
I did that but realized that for an URL looks like this: `"jdbc:postgresql://aawijb5u25wdoy.cpamxznpdoq8.us-west-2.rds.amazonaws.com:5432/ebdb"`, if customers forget to provide `/ebdb` part, then I will treat `aawijb5u25wdoy.cpamxznpdoq8.us-west-2.rds.amazonaws.com:5432` as the db name and `jdbc:postgresql:/` as URL, which is incorrect.